src/System.Text.Json/tests/JsonElementWithNodeParentTests.cs

```diff
@@ -0,0 +1,295 @@
+        }
+
+        [Fact]
+        public static void TestDouble()
+        {
+            Assert.Equal(3.14, new JsonNumber(3.14).AsJsonElement().GetDouble());
+            Assert.Throws<InvalidOperationException>(() => new JsonString().AsJsonElement().GetDouble());
+        }
+
+        [Fact]
+        public static void TestSByte()
+        {
+            Assert.Equal(sbyte.MaxValue, new JsonNumber(sbyte.MaxValue).AsJsonElement().GetSByte());
+            Assert.Throws<InvalidOperationException>(() => new JsonString().AsJsonElement().GetSByte());
+        }
+
+        [Fact]
+        public static void TestUInt16()
+        {
+            Assert.Equal(ushort.MaxValue, new JsonNumber(ushort.MaxValue).AsJsonElement().GetUInt16());
+            Assert.Throws<InvalidOperationException>(() => new JsonString().AsJsonElement().GetUInt16());
+        }
+
+        [Fact]
+        public static void TestUInt32()
+        {
+            Assert.Equal(uint.MaxValue, new JsonNumber(uint.MaxValue).AsJsonElement().GetUInt32());
+            Assert.Throws<InvalidOperationException>(() => new JsonString().AsJsonElement().GetUInt32());
+        }
+
+        [Fact]
+        public static void TestUInt64()
+        {
+            Assert.Equal(ulong.MaxValue, new JsonNumber(ulong.MaxValue).AsJsonElement().GetUInt64());
+            Assert.Throws<InvalidOperationException>(() => new JsonString().AsJsonElement().GetUInt64());
+        }
+
+        [Fact]
+        public static void TestDecimal()
+        {
+            Assert.Equal(decimal.One, new JsonNumber(decimal.One).AsJsonElement().GetDecimal());
+            Assert.Throws<InvalidOperationException>(() => new JsonString().AsJsonElement().GetDecimal());
+        }
+
+        [Fact]
+        public static void TestDateTime()
+        {
+            var dateTime = new DateTime(2019, 1, 1);
+            Assert.Equal(dateTime, new JsonString(dateTime).AsJsonElement().GetDateTime());
+            Assert.Throws<InvalidOperationException>(() => new JsonBoolean().AsJsonElement().GetDateTime());
+        }
+
+        [Fact]
+        public static void TestDateOffset()
+        {
+            var dateTimeOffset = DateTimeOffset.ParseExact("2019-01-01T00:00:00", "s", CultureInfo.InvariantCulture);
+            Assert.Equal(dateTimeOffset, new JsonString(dateTimeOffset).AsJsonElement().GetDateTimeOffset());
+            Assert.Throws<InvalidOperationException>(() => new JsonBoolean().AsJsonElement().GetDateTimeOffset());
+        }
+
+        [Fact]
+        public static void TestGuid()
+        {
+            Guid guid = Guid.ParseExact("ca761232-ed42-11ce-bacd-00aa0057b223", "D");
+            Assert.Equal(guid, new JsonString(guid).AsJsonElement().GetGuid());
+            Assert.Throws<InvalidOperationException>(() => new JsonBoolean().AsJsonElement().GetGuid());
+        }
+
+        [Fact]
+        public static void TestGetRawText()
+        {
+            string jsonToParse = @"{""property name"":""value""}";
```

Comment:
GetRawText will change escaped characters to their unescaped equivalents in that case. It isn't actually returning "raw text" for JsonNode parent, because we don't have it anymore, but rather ToJsonString result (which is pointed out as a remark).